packages/search/src/snippet-index.ts

```diff
@@ -60,7 +60,10 @@ export type SnippetId = {
 export function fileChunkSnippetId(filePath: string, startLine?: number): SnippetId {
   return {
     type: 'file-chunk',
-    id: [filePath, startLine].filter(Boolean).join(':'),
+    id: [filePath, startLine]
+      .filter((t): t is string | number => Boolean(t))
```

Comment:
Can you describe how the old behavior was causing the drive letter bug?

Reply:
Sure, given the following snippet identifier via `[filePath, startLine]`:
```
> ['C:\\Users\\Dustin\\README.md', '1'].filter(Boolean).join(':')
'C:\\Users\\Dustin\\README.md:1'
```

When this string is split into `filePath`, `startLine` below in `parseFileChunkSnippetId`:
```
> 'C:\\Users\\Dustin\\README.md:1'.split(':')
[ 'C', '\\Users\\Dustin\\README.md', '1' ]
```

We then take the first element as the filePath and the second as the startLine, resulting in the path being `C`.
URL encoding the filepath prevents a colon in the path from breaking this process.

Reply:
I'll add a test case for this.

Reply:
Got it, thanks.